### PR TITLE
ServiceComponentManifestCheck extended.

### PR DESCRIPTION
**Pull request by svilenvul**
Extended to verify that all services are included in the build.properties
file.

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

File: src/main/java/org/openhab/tools/analysis/checkstyle/ServiceComponentManifestCheck.java [hunk 1]
```diff
@@ -55,29 +63,96 @@
...
     public void finishProcessing() {
         verifyManifestWildcardDeclaredServiceComponents();
         verifyManifestExplicitlyDeclaredServices();
+        verifyBuildPropertiesFile();
+    }
+
+    private void verifyBuildPropertiesFile() {
+        if (buildPropertiesPath != null) {
+            IBuildEntry binIncludes = buildPropertiesFile.getEntry(IBuildEntry.BIN_INCLUDES);
+            String[] includedTokens = binIncludes.getTokens();
```
Comment (svilenvul) [inline src/main/java/org/openhab/tools/analysis/checkstyle/ServiceComponentManifestCheck.java]: Not null safe.

**Comment by svilenvul** [comment]
@kaikreuzer , I guess you have added the Travis build, please take a look when you have time:

>rake aborted!
No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
/home/travis/.rvm/gems/ruby-1.9.3-p551/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-1.9.3-p551/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)

>travis_time:end:2575e59f:start=1493100985495514675,finish=1493100985605167776,duration=109653101
[0K
[31;1mThe command "rake" exited with 1.[0m

>Done. Your build exited with 1.

**Comment by kaikreuzer** [comment]
I probably did one checkbox too much when activating Travis for openHAB2 - I deactivated it on this repo here again now!

**Comment by kaikreuzer** [comment]
Probably time to do a new release of the tool soon to get all the improvements in production at openhab2-addons, wdyt? Let me know when you think it is a good time and I'll deal with it.

**Comment by svilenvul** [comment]
Great, I think it is a good time for that.

**Comment by svilenvul** [comment]
I have ran the OH2-addons build with the latest changes in the sat repo, the checks are passing, only the cometvisu bundles cause errors in checkstyle:
``` [Fatal Error] pom.xml:1:42: Invalid encoding name "MACROMAN". ```

**Comment by martinvw** [comment]
Interesting! And to what file / line content can it be accounted, I could not find it (on my phone)

**Comment by svilenvul** [comment]
More details:

```
[ERROR] Failed to execute goal org.openhab.tools:static-code-analysis:0.0.5-SNAPSHOT:checkstyle (default) on project org.openhab.ui.cometvisu: Unable to execute
 mojo: An error has occurred in Checkstyle report generation. Failed during checkstyle configuration: Exception was thrown while processing C:\prj\openHAB\Eclip
seIDE\git\openhab2-addons\addons\ui\org.openhab.ui.cometvisu\pom.xml: Unable to read from file: C:\prj\openHAB\EclipseIDE\git\openhab2-addons\addons\ui\pom.xml:
 Invalid encoding name "MACROMAN". -> [Help 1]
```

The same was reported for org.openhab.ui.cometvisu.php bundle.

**Comment by martinvw** [comment]
I would say this is related:

https://github.com/openhab/openhab2-addons/search?utf8=%E2%9C%93&q=MacRoman&type=

**Comment by svilenvul** [comment]
Thanks.

**Pull request merged**